lib/rubocop/ast/node/case_match_node.rb

```diff
@@ -31,6 +31,19 @@ def in_pattern_branches
         node_parts[1...-1]
       end
 
+      # Returns an array of all the when branches in the `case` statement.
+      #
+      # @return [Array<Node, nil>] an array of the bodies of the `in` branches
+      # and the `else` (if any). Note that these bodies could be nil.
+      def branches
+        bodies = in_pattern_branches.map(&:body)
+        if else?
+          # `empty-else` node sets nil because it has no body.
+          else_branch&.empty_else_type? ? bodies.push(nil) : bodies.push(else_branch)
```

Comment:
There seem to be too many conditions here.
1) else?
2) else_branch&.
3) empty_else_type? 

You'd need 4 different tests to cover this.

Reply:
Thank you for your review. The test cases  existed, but the implementation is redundant. I updated this PR.

Reply:
Oops! I made a mistake in a test case, so I updated this PR again.